ERC20-warmup/jagranata/IC3Token.sol

```diff
@@ -0,0 +1,83 @@
+pragma solidity ^0.4.10;
+
+import "./SafeMathLib.sol";
+
+contract IC3Token {
+
+  using SafeMathLib for uint256;
+
+  uint8 public constant decimals = 18;
+  string public constant name = "IC3 2017 Bootcamp Token";
+  string public constant symbol = "IC3";
+
+  uint256 public totalSupply;
+
+  mapping (address => uint256) public balances; // token balances
+  mapping (address => mapping (address => uint256)) public approvals; // approval allotments
+
+  event Transfer(address indexed _from, address indexed _to, uint256 _value);
+  event Approval(address indexed _owner, address indexed _spender, uint256 _value);
+
+  function balanceOf(address _owner) constant returns (uint256 balance) {
+    return balances[_owner];
+  }
+
+  /* deposit 1 wei for 1 ic3 */
+  function deposit () payable returns (bool success) {
+    if (msg.value == 0) return false;
```

Comment:
We may want to allow 0-deposits / return true (this is not specified but worth thinking about)